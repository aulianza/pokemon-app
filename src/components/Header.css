header {
    padding-bottom: 2rem;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.header__logo {
    margin-bottom: 1rem;
}

.header__logo > img {
    width: 100%;
    height: auto;
}

nav {
    display: flex;
    justify-content: center;
    width: 100%;
    align-items: center;
    box-sizing: border-box;
}

nav a {
    text-decoration: none;
    color: black;
    display: flex;
    align-items: center;
    padding: 5px;
    height: 30px;
}

.nav-active {
    background-color: #2b74b9;
    box-shadow: 3px 4px 0 0 #ffcb01;
    color: #fff;
}

.nav-item {
    padding: 7px 20px;
    border-radius: 3px;
    transition: 0.2s;
    margin: 0 5px;
}

.nav-item:hover {
    cursor: pointer;
    color: #fff;
    background-color: #333333;
}

.nav-title {
    font-size: 1.2em;
}

@media only screen and (max-width: 768px) {
    .nav-title {
        font-size: 1.1em;
    }

    .nav-item:hover {
        color: #fff;
        background-color: #2b74b9;
    }
}
