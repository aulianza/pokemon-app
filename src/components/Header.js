import React from "react";
import { NavLink } from "react-router-dom";
import "./Header.css";
import Logo from "../assets/images/pokemon-logo.svg";

const Header = () => {
    return (
        <header>
            <div className="header__logo">
                <h1>
                    <img src={Logo} alt="Pokemon App" title="Pokemon App" width="250" height="80"/>
                </h1>
            </div>
            <nav>
                <NavLink className="nav-item" exact activeClassName="nav-active" to="/">
                    <span className="nav-title">Pokemon List</span>
                </NavLink>
                <NavLink
                    className="nav-item"
                    exact
                    activeClassName="nav-active"
                    to="/my-pokemon"
                >
                    <span className="nav-title">My Pokemon</span>
                </NavLink>
            </nav>
        </header>
    );
};

export default Header;
