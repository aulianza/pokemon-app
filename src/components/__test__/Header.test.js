import Header from "../Header";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from "react-router-dom";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("header-component, snapshot", () => {
    act(() => {
        render(
            <Router>
                <Header />
            </Router>,
            container
        );
    });
    expect(container).toMatchSnapshot();
});
