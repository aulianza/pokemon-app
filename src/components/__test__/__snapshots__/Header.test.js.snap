// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header-component, snapshot 1`] = `
<div>
  <div
    class="css-i2iq46"
  >
    <header>
      <div
        class="header__logo"
      >
        <h1>
          <img
            alt="Pokemon App"
            height="80"
            src="pokemon-logo.svg"
            title="Pokemon App"
            width="250"
          />
        </h1>
      </div>
      <nav>
        <a
          aria-current="page"
          class="nav-item nav-active"
          href="/"
        >
          <span
            class="nav-title"
          >
            Pokemon List
          </span>
        </a>
        <a
          class="nav-item"
          href="/my-pokemon"
        >
          <span
            class="nav-title"
          >
            My Pokemon
          </span>
        </a>
      </nav>
    </header>
  </div>
</div>
`;
